### ISSUE TITLE
Improve SetUrl Logic to handle `$URI` and `$URI`.

### ISSUE THREAD

**Issue by joshrosso** [jfrog/jfrog-client-go#707]
**Is your feature request related to a problem? Please describe.**

As a new user, I struggled to determine why I was getting `404`s on my API requests. Turns out, I was setting the URL as:

```go
ac := auth.NewArtifactoryDetails()
ac.SetUrl("https://arctir.jfrog.io/artifactory")
```

And thus the config was resolving as `https://${ORG_NAME}.jfrog.io/artifactoryapi`.

I'd expect [SetUrl](https://github.com/jfrog/jfrog-client-go/blob/ac964bad46f2549ac8d1b50c2b0e3b632cba0f45/auth/servicedetails.go#L129-L131) to handle both of these cases:

https://github.com/jfrog/jfrog-client-go/blob/ac964bad46f2549ac8d1b50c2b0e3b632cba0f45/auth/servicedetails.go#L129-L131

Just figured I'd pass this feedback/suggestion on from the perspective of a new user 😃. If it's not desirable behavior for the project, feel free to close. 

Thanks for your work on maintaining this library!

**Comment by AF250329** [comment]
yes... it took me hours to understand why HTTP request fails.... all because of the missing / at the end

**Comment by sverdlov93** [comment]
Hi @joshrosso @AF250329 
Thanks for reporting this issue.
Our apologies for the inconvenience caused by this.
This [PR](https://github.com/jfrog/jfrog-client-go/pull/887) should fix it.
I will update this thread once the next release will be out.

**Comment by sverdlov93** [comment]
Hi @joshrosso @AF250329
Version 1.36.0 has been released!
We are looking forward to your feedback on it.
